### ISSUE TITLE
make[2]: *** [CMakeFiles/pycxx.dir/bytecode.cpp.o] Error 1

### ISSUE THREAD

**Issue by BaseMax** [zrax/pycdc#304]
```
[root@5-161-122-190 build]# yum install cmake
Loaded plugins: fastestmirror, universal-hooks
Loading mirror speeds from cached hostfile
 * EA4: 63.247.64.58
 * cpanel-plugins: 63.247.64.58
 * base: mirror.arizona.edu
 * epel: fedora.westmancom.com
 * extras: centos.mirrors.hoobly.com
 * updates: centos.hivelocity.net
Mysql-connectors-community                                                                 | 2.6 kB  00:00:00
Mysql-tools-community                                                                      | 2.6 kB  00:00:00
Mysql57-community                                                                          | 2.6 kB  00:00:00
base                                                                                       | 3.6 kB  00:00:00
extras                                                                                     | 2.9 kB  00:00:00
imunify360                                                                                 | 3.0 kB  00:00:00
imunify360-rollout-1                                                                       | 3.0 kB  00:00:00
imunify360-rollout-2                                                                       | 3.0 kB  00:00:00
imunify360-rollout-3                                                                       | 3.0 kB  00:00:00
imunify360-rollout-4                                                                       | 3.0 kB  00:00:00
updates                                                                                    | 2.9 kB  00:00:00
(1/2): imunify360/7/primary_db                                                             | 1.2 MB  00:00:00
(2/2): updates/7/x86_64/primary_db                                                         |  18 MB  00:00:00
Resolving Dependencies
--> Running transaction check
---> Package cmake.x86_64 0:2.8.12.2-2.el7 will be installed
--> Processing Dependency: libarchive.so.13()(64bit) for package: cmake-2.8.12.2-2.el7.x86_64
--> Running transaction check
---> Package libarchive.x86_64 0:3.1.2-14.el7_7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

==================================================================================================================
 Package                     Arch                    Version                          Repository             Size
==================================================================================================================
Installing:
 cmake                       x86_64                  2.8.12.2-2.el7                   base                  7.1 M
Installing for dependencies:
 libarchive                  x86_64                  3.1.2-14.el7_7                   base                  319 k

Transaction Summary
==================================================================================================================
Install  1 Package (+1 Dependent package)

Total download size: 7.4 M
Installed size: 27 M
Is this ok [y/d/N]: y
Downloading packages:
(1/2): libarchive-3.1.2-14.el7_7.x86_64.rpm                                                | 319 kB  00:00:00
(2/2): cmake-2.8.12.2-2.el7.x86_64.rpm                                                     | 7.1 MB  00:00:00
------------------------------------------------------------------------------------------------------------------
Total                                                                              31 MB/s | 7.4 MB  00:00:00
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : libarchive-3.1.2-14.el7_7.x86_64                                                               1/2
  Installing : cmake-2.8.12.2-2.el7.x86_64                                                                    2/2
  Verifying  : cmake-2.8.12.2-2.el7.x86_64                                                                    1/2
  Verifying  : libarchive-3.1.2-14.el7_7.x86_64                                                               2/2

Installed:
  cmake.x86_64 0:2.8.12.2-2.el7

Dependency Installed:
  libarchive.x86_64 0:3.1.2-14.el7_7

Complete!



[root@5-161-122-190 build]# cmake ..
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
CMake Error at CMakeLists.txt:2 (cmake_minimum_required):
  CMake 3.1 or higher is required.  You are running version 2.8.12.2


-- Configuring incomplete, errors occurred!
See also "/usr/src/redis.unixy.net-4.0.0-el7/pycdc/build/CMakeFiles/CMakeOutput.log".



[root@5-161-122-190 build]# cmake3
-bash: cmake3: command not found



[root@5-161-122-190 build]# yum install cmake3
Loaded plugins: fastestmirror, universal-hooks
Loading mirror speeds from cached hostfile
 * EA4: 63.247.64.58
 * cpanel-plugins: 63.247.64.58
 * base: mirror.arizona.edu
 * epel: fedora.westmancom.com
 * extras: centos.mirrors.hoobly.com
 * updates: centos.hivelocity.net
Resolving Dependencies
--> Running transaction check
---> Package cmake3.x86_64 0:3.17.5-1.el7 will be installed
--> Processing Dependency: cmake3-data = 3.17.5-1.el7 for package: cmake3-3.17.5-1.el7.x86_64
--> Processing Dependency: librhash.so.0()(64bit) for package: cmake3-3.17.5-1.el7.x86_64
--> Processing Dependency: libuv.so.1()(64bit) for package: cmake3-3.17.5-1.el7.x86_64
--> Processing Dependency: libzstd.so.1()(64bit) for package: cmake3-3.17.5-1.el7.x86_64
--> Running transaction check
---> Package cmake3-data.noarch 0:3.17.5-1.el7 will be installed
--> Processing Dependency: emacs-filesystem >= 24.3 for package: cmake3-data-3.17.5-1.el7.noarch
---> Package libuv.x86_64 1:1.44.2-1.el7 will be installed
---> Package libzstd.x86_64 0:1.5.2-1.el7 will be installed
---> Package rhash.x86_64 0:1.3.4-2.el7 will be installed
--> Running transaction check
---> Package emacs-filesystem.noarch 1:24.3-23.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

==================================================================================================================
 Package                         Arch                  Version                          Repository           Size
==================================================================================================================
Installing:
 cmake3                          x86_64                3.17.5-1.el7                     epel                9.6 M
Installing for dependencies:
 cmake3-data                     noarch                3.17.5-1.el7                     epel                1.6 M
 emacs-filesystem                noarch                1:24.3-23.el7                    base                 58 k
 libuv                           x86_64                1:1.44.2-1.el7                   epel                145 k
 libzstd                         x86_64                1.5.2-1.el7                      epel                282 k
 rhash                           x86_64                1.3.4-2.el7                      epel                118 k

Transaction Summary
==================================================================================================================
Install  1 Package (+5 Dependent packages)

Total download size: 12 M
Installed size: 34 M
Is this ok [y/d/N]: y
Downloading packages:
(1/6): cmake3-3.17.5-1.el7.x86_64.rpm                                                      | 9.6 MB  00:00:00
(2/6): cmake3-data-3.17.5-1.el7.noarch.rpm                                                 | 1.6 MB  00:00:00
(3/6): libuv-1.44.2-1.el7.x86_64.rpm                                                       | 145 kB  00:00:00
(4/6): libzstd-1.5.2-1.el7.x86_64.rpm                                                      | 282 kB  00:00:00
(5/6): rhash-1.3.4-2.el7.x86_64.rpm                                                        | 118 kB  00:00:00
(6/6): emacs-filesystem-24.3-23.el7.noarch.rpm                                             |  58 kB  00:00:00
------------------------------------------------------------------------------------------------------------------
Total                                                                              24 MB/s |  12 MB  00:00:00
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : libzstd-1.5.2-1.el7.x86_64                                                                     1/6
  Installing : rhash-1.3.4-2.el7.x86_64                                                                       2/6
  Installing : 1:emacs-filesystem-24.3-23.el7.noarch                                                          3/6
  Installing : 1:libuv-1.44.2-1.el7.x86_64                                                                    4/6
  Installing : cmake3-data-3.17.5-1.el7.noarch                                                                5/6
  Installing : cmake3-3.17.5-1.el7.x86_64                                                                     6/6
  Verifying  : cmake3-3.17.5-1.el7.x86_64                                                                     1/6
  Verifying  : 1:libuv-1.44.2-1.el7.x86_64                                                                    2/6
  Verifying  : 1:emacs-filesystem-24.3-23.el7.noarch                                                          3/6
  Verifying  : rhash-1.3.4-2.el7.x86_64                                                                       4/6
  Verifying  : libzstd-1.5.2-1.el7.x86_64                                                                     5/6
  Verifying  : cmake3-data-3.17.5-1.el7.noarch                                                                6/6

Installed:
  cmake3.x86_64 0:3.17.5-1.el7

Dependency Installed:
  cmake3-data.noarch 0:3.17.5-1.el7     emacs-filesystem.noarch 1:24.3-23.el7     libuv.x86_64 1:1.44.2-1.el7
  libzstd.x86_64 0:1.5.2-1.el7          rhash.x86_64 0:1.3.4-2.el7

Complete!




[root@5-161-122-190 build]# cmake3 ..
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonInterp: /usr/bin/python (found version "2.7.5")
-- Configuring done
-- Generating done
-- Build files have been written to: /usr/src/redis.unixy.net-4.0.0-el7/pycdc/build
[root@5-161-122-190 build]# make
[  2%] Generating bytes/python_10.cpp, bytes/python_11.cpp, bytes/python_13.cpp, bytes/python_14.cpp, bytes/python_15.cpp, bytes/python_16.cpp, bytes/python_20.cpp, bytes/python_21.cpp, bytes/python_22.cpp, bytes/python_23.cpp, bytes/python_24.cpp, bytes/python_25.cpp, bytes/python_26.cpp, bytes/python_27.cpp, bytes/python_30.cpp, bytes/python_31.cpp, bytes/python_32.cpp, bytes/python_33.cpp, bytes/python_34.cpp, bytes/python_35.cpp, bytes/python_36.cpp, bytes/python_37.cpp, bytes/python_38.cpp, bytes/python_39.cpp, bytes/python_310.cpp
Scanning dependencies of target pycxx
[  4%] Building CXX object CMakeFiles/pycxx.dir/bytecode.cpp.o
In file included from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h:4:0,
                 from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/bytecode.cpp:1:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_object.h: In constructor ‘PycObject::PycObject(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_object.h:132:40: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
     PycObject(int type = TYPE_UNKNOWN) : m_refs(0), m_type(type) { }
                                        ^
In file included from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/bytecode.cpp:1:0:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h: In constructor ‘PycInt::PycInt(int, int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h:12:9: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
         : PycObject(type), m_value(value) { }
         ^
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h:12:9: error: declaration of ‘value’ shadows a member of 'this' [-Werror=shadow]
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h: In constructor ‘PycLong::PycLong(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h:31:9: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
         : PycObject(type), m_size(0) { }
         ^
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h: In constructor ‘PycFloat::PycFloat(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h:50:9: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
         : PycObject(type) { }
         ^
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h: In constructor ‘PycComplex::PycComplex(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h:65:9: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
         : PycFloat(type) { }
         ^
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h: In constructor ‘PycCFloat::PycCFloat(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h:80:9: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
         : PycObject(type), m_value(0.0) { }
         ^
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h: In constructor ‘PycCComplex::PycCComplex(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_numeric.h:99:9: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
         : PycCFloat(type), m_imag(0.0) { }
         ^
In file included from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_code.h:4:0,
                 from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/bytecode.h:1,
                 from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/bytecode.cpp:2:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_sequence.h: In constructor ‘PycSequence::PycSequence(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_sequence.h:10:27: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
     PycSequence(int type) : PycObject(type), m_size(0) { }
                           ^
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_sequence.h: In constructor ‘PycTuple::PycTuple(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_sequence.h:23:37: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
     PycTuple(int type = TYPE_TUPLE) : PycSequence(type) { }
                                     ^
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_sequence.h: In constructor ‘PycList::PycList(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_sequence.h:40:35: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
     PycList(int type = TYPE_LIST) : PycSequence(type) { }
                                   ^
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_sequence.h: In constructor ‘PycDict::PycDict(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_sequence.h:58:35: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
     PycDict(int type = TYPE_DICT) : PycSequence(type) { }
                                   ^
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_sequence.h: In constructor ‘PycSet::PycSet(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_sequence.h:79:33: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
     PycSet(int type = TYPE_SET) : PycSequence(type) { }
                                 ^
In file included from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_code.h:5:0,
                 from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/bytecode.h:1,
                 from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/bytecode.cpp:2:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_string.h: In constructor ‘PycString::PycString(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_string.h:12:9: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
         : PycObject(type) { }
         ^
In file included from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/bytecode.h:1:0,
                 from /usr/src/redis.unixy.net-4.0.0-el7/pycdc/bytecode.cpp:2:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_code.h: In constructor ‘PycCode::PycCode(int)’:
/usr/src/redis.unixy.net-4.0.0-el7/pycdc/pyc_code.h:32:9: error: declaration of ‘type’ shadows a member of 'this' [-Werror=shadow]
         : PycObject(type), m_argCount(), m_posOnlyArgCount(), m_kwOnlyArgCount(),
         ^
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/pycxx.dir/bytecode.cpp.o] Error 1
make[1]: *** [CMakeFiles/pycxx.dir/all] Error 2
make: *** [all] Error 2




[root@5-161-122-190 build]# ls
bytes  CMakeCache.txt  CMakeFiles  cmake_install.cmake  Makefile
```

**Comment by UnoArroefy** [comment]
have you resolved this?? i also have the same issue

**Comment by bashidi** [comment]
have you resolved this?i get the same issue too in my project :(

**Comment by greenozon** [comment]
this error is because Centos 6/7 uses very old gcc/g++ compiler which has some false positives

in order to fix above set of issues one has to remove 
-Werror=shadow

from CMakefile

**Comment by zrax** [comment]
Fixed with #469
